src/core/strong_typedef_type.hpp

```diff
@@ -127,8 +127,8 @@ namespace StrongType {
 			friend constexpr bool operator >=(const TType &lhs, TCompatibleType rhs) { return lhs.value >= static_cast<TBaseType>(rhs); }
 			friend constexpr bool operator >(const TType &lhs, TCompatibleType rhs) { return lhs.value > static_cast<TBaseType>(rhs); }
 
-			friend constexpr TType operator +(const TType &lhs, TCompatibleType rhs) { return { static_cast<TBaseType>(lhs.value + rhs) }; }
-			friend constexpr TType operator -(const TType &lhs, TCompatibleType rhs) { return { static_cast<TBaseType>(lhs.value - rhs) }; }
+			friend constexpr TType operator +(const TType &lhs, TCompatibleType rhs) { return TType{ static_cast<TBaseType>(lhs.value + rhs) }; }
+			friend constexpr TType operator -(const TType &lhs, TCompatibleType rhs) { return TType{ static_cast<TBaseType>(lhs.value - rhs) }; }
```

Comment:
I think the usual pattern should be something like
```c++
friend constexpr TType operator +(TType lhs, TCompatibleType rhs) { lhs += value; return lhs }
```
which of course uses the overloaded `+=` operator.

Reply:
There is no `+=` operator for `TCompatibleType` though.

Reply:
Oh. Fair enough!